Only perform migration events on living simulants

zmbc opened the pull request:
## Only perform migration events on living simulants

### Description
<!-- For use in commit message, wrap at 72 chars. 72 chars is here: -->
- *Category*: bugfix
- *JIRA issue*: [MIC-3723](https://jira.ihme.washington.edu/browse/MIC-3723)
- *Research reference*: none

### Changes and notes

### Verification and Testing

Verified simulation still runs, and that this call returns a slightly smaller dataframe than before.

rmudambi reviewed (approved):
Can you also include the prohibition on dead people changing jobs?

Also include tests for this if possible.

zmbc commented:
> Can you also include the prohibition on dead people changing jobs?

See #126, I figured that would be merged directly to main so I separated them.

zmbc commented:
@rmudambi I've added a test, but because this test needs to look at consecutive time steps, I have refactored the testing setup to make that easier. Do you think those changes are appropriate to include in this PR or should I extract them? I will need this capability for most migration tests, I think.

rmudambi commented on integration_tests/test_domestic_migration.py:
```diff
@@ -0,0 +1,39 @@
+        after = after.loc[common_simulants]
+
+        movers = before["address_id"] != after["address_id"]
+        assert after[movers]["tracked"].all()
```
What if there were no dead people at all? You should probably assert that there are some people that are dead who could have moved.

zmbc replied:
Hmm... I was thinking that the test here was that the component only moves living simulants; that people die in the sim is a pretty basic assumption that IMO would go in a test of the mortality component. But, if that doesn't sound right to you I can add the assert here.

rmudambi replied:
No I think you're right, we should test that independently
